tests/tests.py

```diff
@@ -250,3 +250,44 @@ def test_decimal_ignore_order(self):
         result = {}
         self.assertEqual(ddiff, result)
 
+    def test_unicode_string_type_changes(self):
+        unicodeString = {"hello": u"你好"}
+        asciiString = {"hello": "你好"}
+        ddiff = DeepDiff(unicodeString, asciiString)
+        if type(unicodeString["hello"]) == str:
+            # In python3, all string is unicode, so diff is empty
+            result = {}
+        else:
+            # In python2, these are 2 different type of strings
+            result = {'type_changes': [u"root['hello']: \u4f60\u597d=<type 'unicode'> ===> \ufffd\ufffd\ufffd\ufffd\ufffd\ufffd=<type 'str'>"]}
+        self.assertEqual(ddiff, result)
+
+    def test_unicode_string_value_changes(self):
+        unicodeString = {"hello": u"你好"}
+        asciiString = {"hello": u"你好hello"}
+        ddiff = DeepDiff(unicodeString, asciiString)
+        if type(unicodeString["hello"]) == str:

```

Comment:
Great test cases. I would change the `if type(unicodeString["hello"]) == str:` to `if py3:`
We can add a few lines at the top of the file to detect py version:

```python
from sys import version
py3 = version[0] == '3'
```